final_task/rss_reader/Handler.py

```diff
@@ -0,0 +1,260 @@
+            self.write_cache(self.convert_Entry_to_dict(entry))
+        self.write_to_pdf(path)
+
+    @logging_decorator
+    def option_json(self) -> None:
+        for entry in self.entries:
+            self.write_cache(self.convert_Entry_to_dict(entry))
+            self.print_to_json(self.convert_Entry_to_dict(entry))
+
+    @logging_decorator
+    def option_default(self) -> None:
+        for entry in self.entries:
+            self.print_entry(entry)
+            self.write_cache(self.convert_Entry_to_dict(entry))
+
+    @logging_decorator
+    def correct_title(self, title: str) -> str:
+        return title.replace('"', "_").replace("?", "_").replace(":", "_").replace("'", "_")[:30].replace(" ", "_")
+
+    @logging_decorator
+    def write_cache(self, entry_dict: dict):
+        """write new news to cache in json format"""
+        try:
+            entries = json.load(open("cache.json"))
+        except json.JSONDecodeError:
+            entries = []
+        except FileNotFoundError:
+            entries = []
+        # writing entries and saving images of entries to cache if list of entries with this title is empty
+        # else this entry already is the cache
+        if not [entry for entry in entries if entry["Title"] == entry_dict["Title"]]:
+            entries.append(entry_dict)
+            if len(entry_dict["Links"]) > 1:
+                self.save_image(entry_dict["Links"][1], self.correct_title(entry_dict["Title"]))
+
+        with open("cache.json", "w") as cache:
+            json.dump(entries, cache, indent=2)
+
+    @logging_decorator
+    def save_image(self, img_url: str, img_name: str):
+        try:
+            if img_name.find(".jpg") == -1:
+                urllib.request.urlretrieve(img_url, f"images/{img_name}.jpg")
+            else:
+                urllib.request.urlretrieve(img_url, f"images/{img_name}")
+        except FileNotFoundError:
+            os.makedirs("images")
+            urllib.request.urlretrieve(img_url, f"images/{img_name}.jpg")
+
+    @logging_decorator
+    def option_date(self, date: str, do_json: bool, do_html: bool, html_path: str, do_pdf: bool, pdf_path: str):
+        """add entries from cache.json into daily_news: list if they have date that is equal to user's --date DATE
+            and then raise an exception or print to console or to outputs to html"""
+        try:
+            entries = json.load(open("cache.json"))
+        except json.JSONDecodeError:
+            entries = []
+        except FileNotFoundError:
+            raise RSSReaderException("Error. You have no cache. Try to run app with internet-connection")
+
+        # list of entries with the same date as user's --date DATE
+        daily_news = [entry for entry in entries if entry["DateInt"] == date]
+        if not daily_news:
+            raise RSSReaderException("Error. News aren't found")
+        # different cases of command line arguments
+        elif do_pdf and do_html:
```

Comment:
You can optimize this statements. Instead of elif, you could just check your arguments one by one with if and do some logic, instead of duplicating your code.